File: src/main/java/com/dersaun/apigestaocontas/api/controllers/DashboardController.java
package com.dersaun.apigestaocontas.api.controllers;

import com.dersaun.apigestaocontas.domain.dtos.GastosFormaPagamentoMesAnoDTO;
import com.dersaun.apigestaocontas.domain.dtos.MesAnoDTO;
import com.dersaun.apigestaocontas.domain.dtos.ResumoFaturaPessoas;
import com.dersaun.apigestaocontas.domain.dtos.ResumoFormaPagamentoMensalResponseDTO;
import com.dersaun.apigestaocontas.domain.services.DashboardService;
import com.dersaun.apigestaocontas.domain.services.ResumoFaturaService;
import org.eclipse.collections.api.list.MutableList;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("dashboard")
public class DashboardController {

    @Autowired
    private ResumoFaturaService resumoFaturaService;

    @Autowired
    private DashboardService dashboardService;

    @GetMapping("resumo-pagamentos/{pessoaId}")
    public ResponseEntity<ResumoFormaPagamentoMensalResponseDTO> resumoPagamentos(
            @PathVariable("pessoaId") Long pessoaId,
            MesAnoDTO mesAnoDTO
    ) {
        var resumos = resumoFaturaService.buscarResumoPagamentos(mesAnoDTO, pessoaId);

        return ResponseEntity.ok(resumos);
    }

    @GetMapping("resumo-fatura-pessoas")
    public ResponseEntity<List<ResumoFaturaPessoas>> resumoFaturaPessoas(MesAnoDTO mesAnoDTO) {
        var resumos = resumoFaturaService.resumoFaturaPessoas(mesAnoDTO);
        return ResponseEntity.ok(resumos);
    }

    @GetMapping("grafico/gastos-por-pessoa")
    public ResponseEntity<?> graficoGastosPorPessoa(MesAnoDTO mesAnoDTO) {
        var resultado = this.dashboardService.somarValoresCadaPessoaMesesAnteriores(
                mesAnoDTO, 8, true
        );
        return ResponseEntity.ok(resultado);
    }

    @GetMapping("grafico/gastos-por-cartao")
    public ResponseEntity<MutableList<GastosFormaPagamentoMesAnoDTO>> dadosGraficoGastorPorFormaPagamento(MesAnoDTO mesAnoDTO) {
        var resultado = this.dashboardService.criarGraficoFormaPagamento(
                mesAnoDTO, 4, true
        );
        return ResponseEntity.ok(resultado);
    }
}

File: src/app/dashboard/controllers/dashboard.controller.ts
import { UsuarioAutenticadoDto } from '@app/autenticacao/dtos/usuario-autenticado.dto';
import { MesAnoDTO } from '@app/conta/dtos/mes-ano.dto';
import { ResumoFaturaPessoas } from '@app/conta/dtos/resumo-faturas';
import { ResumoFormaPagamentoMensalResponseDTO } from '@app/conta/dtos/resumo-forma-pagamento-mensal-response.dto';
import { IResumoFaturaService } from '@app/conta/interfaces/resumo-fatura.service';
import { GastosFormaPagamentoMesAnoDTO } from '@app/dashboard/dtos/gastos-forma-pagamento-mes-ano.dto';
import { GastosPessoaMesAnoDTO } from '@app/dashboard/dtos/gastos-pessoa-mes-ano.dto';
import { Pessoa } from '@app/pessoa/models/pessoa.entity';
import { User } from '@decorators/usuario.decorator';
import { Controller, Get, Inject, Param, Query } from '@nestjs/common';
import { IDashboardService } from '@app/dashboard/interfaces/dashboard.service.interface';

@Controller({ version: '1', path: 'dashboard' })
export class DashboardController {
  constructor(
    @Inject('IResumoFaturaService')
    private readonly resumoFaturaService: IResumoFaturaService,
    @Inject('IDashboardService')
    private readonly dashboardService: IDashboardService,
  ) {}

  @Get('resumo-pagamentos/:pessoaId')
  async resumoPagamentos(
    @Param('pessoaId') pessoaId: number,
    @Query() mesAnoDTO: MesAnoDTO,
  ): Promise<ResumoFormaPagamentoMensalResponseDTO> {
    const resumos = await this.resumoFaturaService.buscarResumoPagamentos(
      mesAnoDTO,
      pessoaId,
    );
    return resumos;
  }

  @Get('resumo-fatura-pessoas')
  async resumoFaturaPessoas(
    @Query() mesAnoDTO: MesAnoDTO,
    @User() usuario: UsuarioAutenticadoDto,
  ): Promise<ResumoFaturaPessoas[]> {
    const resumos = await this.resumoFaturaService.resumoFaturaPessoas(
      mesAnoDTO,
      usuario,
    );
    return resumos;
  }
  @Get('grafico/gastos-por-pessoa')
  async graficoGastosPorPessoa(
    @Query() mesAnoDTO: MesAnoDTO,
    @User() usuario: UsuarioAutenticadoDto,
  ): Promise<GastosPessoaMesAnoDTO[]> {
    const resultado =
      await this.dashboardService.somarValoresCadaPessoaMesesAnteriores(
        mesAnoDTO,
        8,
        true,
        usuario.ifNotAdminGetPessoa(),
      );
    return resultado;
  }
  @Get('grafico/gastos-por-pessoa/:pessoaId')
  async graficoGastosPorPessoaPorId(
    @Param('pessoaId') pessoaId: number,
    @Query() mesAnoDTO: MesAnoDTO,
  ): Promise<GastosPessoaMesAnoDTO[]> {
    const resultado =
      await this.dashboardService.somarValoresCadaPessoaMesesAnteriores(
        mesAnoDTO,
        5,
        true,
        { id: pessoaId } as Pessoa,
      );
    return resultado;
  }
  @Get('grafico/gastos-por-cartao')
  async dadosGraficoGastorPorFormaPagamento(
    @Query() mesAnoDTO: MesAnoDTO,
  ): Promise<GastosFormaPagamentoMesAnoDTO[]> {
    const resultado = await this.dashboardService.criarGraficoFormaPagamento(
      mesAnoDTO,
      4,
      true,
    );
    return resultado;
  }
}
